sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java

```diff
@@ -134,6 +134,12 @@ public PCollection<Row> expand(PBegin input) {
             if (partitions != null) {
               readRows = readRows.withNumPartitions(partitions);
             }
+
+            @Nullable Short fetchSize = config.getInt16("fetchSize");
```

Comment:
Yes, I found it strange but just replicated what was in the non-partitioned.

I think making it clearer and switching the default to int16_max instead of 50000 would make more sense, and make JdbcIO more usable / drop-in for most databases (since we reportedly had issues when some databases went over it).

What do you think?

Reply:
this would change existing behavior. If change the default we should switch the int16 to int32 type for fetchSize everywhere (SchemaIOProvider, Python wrapper) to give an option for user to keep the original size (50,000). Current change also LGTM.